### PR TITLE
Complete calendar walkthrough assignment part 1.

### PR DESCRIPTION
**Pull request by mbouadeus**
Complete calendar assignment part I for week-5 walkthroughs.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (isaacwdavis) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit: add space before colon "meetingAttendee : meetingAttendees"

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (isaacwdavis) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: nit: add period to this comment and the comments below

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (pengxianzhe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:29]: Put an empty line between the general description and first @param.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (pengxianzhe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Are you sure you need to have a recursive method? This method is just looping through all events in the event list, so it is not really a recursive algorithm.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (pengxianzhe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:47]: I think you can just do something similar to Collections.sort(eventList, TimeRange.ORDER_BY_START);

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (pengxianzhe) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: You need your event list to be sorted for this method to work. I see your sortEventByStart method, but you are not calling it anywhere. Am I missing something?

**Review by isaacwdavis — COMMENTED** [review]
Where are the tests?

Also address David's comments.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Yes, I agree. I'll make it iterative.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:47]: I can't because the comparitor TimeRange.ORDER_BY_START is for TimeRange objects not Event objects. Event doesn't have a comparitor of its own so I think I have to make one.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Oh, great catch. I didn't even notice it. I should write a test that doesn't have the events already sorted by starting times.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: > Where are the tests?
> 
> Also address David's comments.

The tests were already written for us. We write our own test on part 2 of the assignment. I was waiting for this pull request to be merged before making making a pull request for part 2.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Fixed it

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Fixed it.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Fixed it

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (mbouadeus) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:29]: Fixed it.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java
Comment (pengxianzhe) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:47]: SGTM.

**Review by pengxianzhe — COMMENTED** [review]
You didn't push your fixes into this pull request yet. Once you do that I will approve this pull request.

**Review by isaacwdavis — APPROVED** [review]

**Pull request merged**
